code/modules/paperwork/paper_bundle.dm

```diff
@@ -136,53 +127,45 @@
 		+ "</body></html>", "window=[name]")
 
 /obj/item/paper_bundle/attack_self(mob/user as mob)
-	src.show_content(user)
+	show_content(user, opened_page)
 	add_fingerprint(usr)
 	update_icon()
 	return
 
 /obj/item/paper_bundle/Topic(href, href_list)
 	..()
-	if((src in usr.contents) || (istype(src.loc, /obj/item/folder) && (src.loc in usr.contents)))
+	var/can_flip = (src in usr.contents) || (istype(loc, /obj/item/folder) && (loc in usr.contents))
+	if(can_flip || istype(usr, /mob/dead/observer))
 		usr.set_machine(src)
-		if(href_list["next_page"])
-			if(page == amount)
-				screen = 2
-			else if(page == 1)
-				screen = 1
-			else if(page == amount+1)
-				return
-			page++
-			playsound(src.loc, "pageturn", 50, 1)
-		if(href_list["prev_page"])
-			if(page == 1)
-				return
-			else if(page == 2)
-				screen = 0
-			else if(page == amount+1)
-				screen = 1
-			page--
-			playsound(src.loc, "pageturn", 50, 1)
-		if(href_list["remove"])
-			var/obj/item/W = src[page]
+		var/viewed_page = text2num(href_list["cur_page"])
+		if(href_list["flip"])
+			var/flip_count = text2num(href_list["flip"])
+			if(!isnum(flip_count))  // should never happen
+				stack_trace("Paper bundle was flipped without specifying numeric flip amount. href_list\[\"flip\"] == [href_list["flip"]]")
+				flip_count = 0
+			var/new_page = Clamp(1, viewed_page + flip_count, contents.len)
+			if(can_flip && new_page != opened_page)
+				opened_page = new_page
+				playsound(loc, "pageturn", 50, 1)
+			show_content(usr, new_page)
```

Comment:
This is already done in the attack_self call on line 168 which will override the page it's on with the `opened_page` value.
It works due to the attack_self call only going of for mobs holding it in their contents but this causes more networking and what not due to the double showing of the same window. You do want to keep the finger print stuff though I imagine that attack_hand gives.